root/static/scripts/release-editor/validation.js

```diff
@@ -123,13 +127,45 @@ $(function () {
 });
 
 
+// Search releases with the same barcode
+function searchExistingBarcode(field, barcode) {
+    utils.search("release", `barcode:${barcode}`, 1).done(data => {
+        if (data.releases.length) {
+
+            const msg = l(
+                `The following releases with that barcode are already in the
+                 MusicBrainz database. Please make sure you are not adding an
+                 exact duplicate of any of these:`,
+            );
+            const releaseList = ReactDOMServer.renderToString(
+                <ul>
+                    {data.releases.map(release => {
```

Comment:
This looks great, but I'd just put the msg inside the `renderToString` call (you can do it with a React fragment, `<> </>`) so that any potential HTML in the translated string is escaped there.
```suggestion
            const releaseList = ReactDOMServer.renderToString(
                <>
                    {msg}
                    <ul>
                        {data.releases.map(release => {
```